src/project_insights.py

```diff
@@ -0,0 +1,311 @@
+"""
+project_insights
+================
+
+Utility helpers for persisting and querying analyzed project insights.
+
+Responsibilities:
+- Append analysis output (hierarchy, résumé info, skills, contributors) to a JSON log
+- Provide a chronological listing for projects
+- Rank projects based on contribution signals
+"""
+
+from __future__ import annotations
+
+import json
+import uuid
+from dataclasses import asdict, dataclass, field
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+JsonEntry = Dict[str, Any]
+ContributorData = Dict[str, Dict[str, Any]]
+PathLike = Union[str, Path]
+
+DEFAULT_STORAGE = Path("User_config_files/project_insights.json")
+
+
+def _now_iso(ts: Optional[datetime] = None) -> str:
+    """Return an ISO 8601 timestamp in UTC."""
+    ts = ts or datetime.now(tz=timezone.utc)
+    return ts.astimezone(timezone.utc).isoformat()
+
```

Comment:
```suggestion
def _now_iso(ts: Optional[datetime] = None) -> str:
    """Return an ISO 8601 timestamp in UTC."""
    if ts is None:
        ts = datetime.now(timezone.utc)

    # Ensure timestamp is timezone-aware and in UTC
    if ts.tzinfo is None:
        ts = ts.replace(tzinfo=timezone.utc)

    return ts.astimezone(timezone.utc).isoformat()

```
This addition helps in making sure that your system handles naive datetimes, and forces them to be in UTC safely, something for you modified for next time